lib/reduce_quda.cu

```diff
@@ -240,7 +240,7 @@ namespace quda {
     {
       // in case of x.Ncolor()!=3 (MG mainly) reduce_core do not support this function.
       if (x.Ncolor()!=3) return make_double3(0.0, 0.0, 0.0);
-      auto rtn = instantiateReduce<xpyHeavyQuarkResidualNorm_, false>(0.0, 0.0, 0.0, x, y, r, r, r);
+      auto rtn = instantiateReduce<xpyHeavyQuarkResidualNorm_, true>(0.0, 0.0, 0.0, x, y, r, r, r);
```

Comment:
Completely orthogonal to this PR, but _only_ because we're here---should we make this function `errorQuda` if `x.Ncolor() != 3` evaluates to `true`?

Reply:
The reason for this behavior is so that solvers that call the HQ residual don't barf when being used as a smoother or coarse grid solver.